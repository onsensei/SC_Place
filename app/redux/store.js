import createSagaMiddleware from 'redux-saga';
import rootReducer from './reducers/index.reducer';
import rootSaga from './sagas/index.saga';
import {applyMiddleware, compose, createStore} from 'redux';
import {fork} from 'redux-saga/effects';
// import * as actions from './actions/index.action';

// middleware

const composeEnhancers = global.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ? global.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({}) : compose;

const sagaMiddleware = createSagaMiddleware();

const greetingMiddleware = () => (next) => (action) => {
  // if (action.type === actions.GOOD_NIGHT_ACTION) {
  //   action.payload = 'Sweet Dream';
  // }

  next(action);
};

const composedEnhancer = composeEnhancers(applyMiddleware(sagaMiddleware, greetingMiddleware));

function* sagas () {
  yield fork(rootSaga);
}

// store

export const initStore = () => {
  const store = createStore(rootReducer, {}, composedEnhancer);
  sagaMiddleware.run(sagas);
  return store;
};
