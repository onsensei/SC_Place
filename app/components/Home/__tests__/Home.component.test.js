import ConnectedHome, {mapDispatchToProps} from '../Home.component';

// Note: test renderer must be required after react-native.
import React from 'react';
import renderer from 'react-test-renderer';

import {createStore} from 'redux';
import {NavigationActions} from 'react-navigation';
import {shallow} from 'enzyme';

// constant
const store = createStore(() => ({}));

describe('Home', () => {
  let connectedComponent;
  let wrapper;
  let instance;

  beforeEach(() => {
    connectedComponent = <ConnectedHome store={store}/>;

    const connectedWrapper = shallow(connectedComponent);
    wrapper = connectedWrapper.find('Home').shallow();
    instance = wrapper.instance();
  });

  it('renders correctly', () => {
    const tree = renderer.create(connectedComponent);
    expect(tree).toMatchSnapshot();
  });
  
  it('onPressCountButton should increase count', () => {
    // arrange
    instance.setState({
      count: 0
    });

    // act & assert
    instance.onPressCountButton();
    expect(instance.state.count).toEqual(1);

    instance.onPressCountButton();
    expect(instance.state.count).toEqual(2);

    instance.onPressCountButton();
    expect(instance.state.count).toEqual(3);
  });

  it('onPressGoodMorningButton should call goodMorningDispatcher with params', () => {
    // arrange
    wrapper.setProps({
      goodMorningDispatcher: jest.fn()
    });
    instance = wrapper.instance();

    // act
    instance.onPressGoodMorningButton();

    // assert
    const params = {
      message: 'Good Morning from dispatcher',
      time: '8:00'
    };
    expect(instance.props.goodMorningDispatcher).toHaveBeenCalledWith(params);
  });

  it('onPressGoodAfternoonButton should call goodAfternoonDispatcher with params', () => {
    // arrange
    wrapper.setProps({
      goodAfternoonDispatcher: jest.fn()
    });
    instance = wrapper.instance();

    // act
    instance.onPressGoodAfternoonButton();

    // assert
    const params = {
      message: 'Good Afternoon from dispatcher',
      time: '14:00'
    };
    expect(instance.props.goodAfternoonDispatcher).toHaveBeenCalledWith(params);
  });

  it('onPressGoodNightButton should call goodNightDispatcher with params', () => {
    // arrange
    wrapper.setProps({
      goodNightDispatcher: jest.fn()
    });
    instance = wrapper.instance();

    // act
    instance.onPressGoodNightButton();

    // assert
    const params = {
      message: 'Good Night from dispatcher',
      time: '20:00'
    };
    expect(instance.props.goodNightDispatcher).toHaveBeenCalledWith(params);
  });

  it('onPressGreetingAllButton should call greetingAllDispatcher', () => {
    // arrange
    wrapper.setProps({
      greetingAllDispatcher: jest.fn()
    });
    instance = wrapper.instance();

    // act
    instance.onPressGreetingAllButton();

    // assert
    expect(instance.props.greetingAllDispatcher).toHaveBeenCalled();
  });

  it('onPressGoToListPageButton should call navigateToList', () => {
    // arrange
    wrapper.setProps({
      navigateToList: jest.fn()
    });
    instance = wrapper.instance();

    // act
    instance.onPressGoToListPageButton();

    // assert
    expect(instance.props.navigateToList).toHaveBeenCalled();
  });

  it('navigateToList should call dispatch with params', () => {
    // arrange
    const dispatch = jest.fn();
    const props = mapDispatchToProps(dispatch);

    // act
    props.navigateToList();

    // assert
    expect(dispatch).toHaveBeenCalledWith(NavigationActions.navigate({routeName: 'List'}));
  });
});
