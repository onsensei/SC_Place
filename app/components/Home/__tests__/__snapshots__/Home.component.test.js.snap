// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    To get started, edit Home.component.js
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Press Cmd+R to reload,
Cmd+D or shake for dev menu
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    count : 
    0
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        count++
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Greeting: 
    
     at 
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Good Morning
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Good Afternoon
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Good Night
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Greeting All
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Navigate
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Go to List page &gt;
      </Text>
    </View>
  </View>
</View>
`;
